### ISSUE TITLE
ThrowInt32OverflowException on Issue.Events.GetAllForRepository

### ISSUE THREAD

**Issue by Thraka** [octokit/octokit.net#2031]
Using octokit 0.36.0

The following code causes an exception:

```csharp
var results = await client.Issue.Events.GetAllForRepository("dotnet", "docs");
```

```
This exception was originally thrown at this call stack:
	System.Convert.ThrowInt32OverflowException()
	System.Convert.ToInt32(long)
	long.System.IConvertible.ToInt32(System.IFormatProvider)
	System.Convert.ChangeType(object, System.Type, System.IFormatProvider)
	Octokit.PocoJsonSerializerStrategy.DeserializeObject(object, System.Type) in SimpleJson.cs
	Octokit.Internal.SimpleJsonSerializer.GitHubSerializerStrategy.DeserializeObject(object, System.Type) in SimpleJsonSerializer.cs
	Octokit.PocoJsonSerializerStrategy.DeserializeObject(object, System.Type) in SimpleJson.cs
	Octokit.Internal.SimpleJsonSerializer.GitHubSerializerStrategy.DeserializeObject(object, System.Type) in SimpleJsonSerializer.cs
	Octokit.PocoJsonSerializerStrategy.DeserializeObject(object, System.Type) in SimpleJson.cs
	Octokit.Internal.SimpleJsonSerializer.GitHubSerializerStrategy.DeserializeObject(object, System.Type) in SimpleJsonSerializer.cs
	...
	[Call Stack Truncated]
```

**Comment by shiftkey** [comment]
This is the problem property:

https://github.com/octokit/octokit.net/blob/74dc51a6f567395d0c46d97f7270f959d671573e/Octokit/Models/Response/IssueEvent.cs#L26-L29

And much like we did in #1940, changing it to a `long` should fix the issue.

**Comment by terrance00** [comment]
I'd like to contribute, forked.

**Comment by terrance00** [comment]
Hi @shiftkey, I've got a question.

When we change that model from `int` to `long` for the `Id`, some tests break:

```
    //IssueEventsClientTests.cs
    var issueEvents = await _issuesEventsClient.GetAllForRepository(_context.Repository.Id);
    int issueEventId = issueEvents[0].Id;
```

This does then also break those interfaces it calls.

Shall I follow the chain down and change the interfaces and implementations to accept longs etc.?

Let me know - I'll create a PR by following the chain all the way for the longs.

**Comment by shiftkey** [comment]
> Shall I follow the chain down and change the interfaces and implementations to accept longs etc.?

Yeah, we're gonna have to break those cases too.

**Comment by mijacobs** [comment]
I'm running into this issue as well, is there an ETA for a fix?

**Comment by shiftkey** [comment]
Closing this out as I believe this has been fixed in #2060 and is now available on NuGet.
